src/idl/src/tree.c

```diff
@@ -990,6 +990,34 @@ idl_finalize_struct(
   idl_struct_t *node,
   idl_member_t *members)
 {
+  idl_member_t *member, *member2;
+  idl_declarator_t *decl, *decl2;
+
+  if (node->inherit_spec) {
+    idl_struct_t *base = node->inherit_spec->base;
+    assert(idl_is_struct(base));
+
+    /*check for clashes between inherited and inheriting members*/
+    IDL_FOREACH(member, members) {
+      if (member->key.value) {
+        idl_error(state, idl_location(member),
+          "Derived struct is not allowed to expand on keys");
+        return IDL_RETCODE_SEMANTIC_ERROR;
+      }
+      IDL_FOREACH(decl, member->declarators) {
+        IDL_FOREACH(member2, base->members) {
+          IDL_FOREACH(decl2, member2->declarators) {
+            if (0 == strcmp(decl->name->identifier,decl2->name->identifier)) {
+              idl_error(state, idl_location(decl),
+                "Member name %s clashes with declaration in inheritance", decl->name->identifier);
+              return IDL_RETCODE_SEMANTIC_ERROR;
+            }
+          }
+        }
+      }
+    }
+  }
+
```

Comment:
This should really be implemented in the scoping implementation. There's imports already that check for names etc. I meant to implement it there for inheritance too, but it must've slipped my mind.

Reply:
@k0ekk0ek The spec states that an inheriting struct must either have the same extensibility as its parent or have no explicit extensibility type specified. But I agree that the approach as proposed in section 7.2.2.4.5 in the XTypes spec causes potential backward incompatibility issues when people add attributes to non-mutable structs that are then inherited by other non-mutable structs. I have posted a question about this to the XTypes RTF under ticket DDSXTY14-51.